src/Microsoft.Azure.SignalR.Management/ServiceManager.cs

```diff
@@ -51,54 +51,18 @@ internal ServiceManager(ServiceManagerOptions serviceManagerOptions, string prod
 
         public async Task<IServiceHubContext> CreateHubContextAsync(string hubName, ILoggerFactory loggerFactory = null, CancellationToken cancellationToken = default)
         {
-            loggerFactory = loggerFactory ?? NullLoggerFactory.Instance;
+            loggerFactory ??= NullLoggerFactory.Instance;
             switch (_serviceManagerOptions.ServiceTransportType)
             {
                 case ServiceTransportType.Persistent:
                     {
-                        var connectionFactory = new ManagementConnectionFactory(_productInfo, new ConnectionFactory(_serverNameProvider, loggerFactory));
-                        var serviceProtocol = new ServiceProtocol();
-                        var clientConnectionManager = new ClientConnectionManager();
-                        var clientConnectionFactory = new ClientConnectionFactory();
-                        ConnectionDelegate connectionDelegate = connectionContext => Task.CompletedTask;
-                        var serviceConnectionFactory = new ServiceConnectionFactory(
-                            serviceProtocol,
-                            clientConnectionManager,
-                            connectionFactory,
-                            loggerFactory,
-                            connectionDelegate,
-                            clientConnectionFactory,
-                            new DefaultServerNameProvider()
-                            );
-                        var weakConnectionContainer = new WeakServiceConnectionContainer(
-                            serviceConnectionFactory,
-                            _serviceManagerOptions.ConnectionCount,
-                            new HubServiceEndpoint(hubName, _endpointProvider, _endpoint),
-                            loggerFactory?.CreateLogger(nameof(WeakServiceConnectionContainer)) ?? NullLogger.Instance);
-
-                        var serviceCollection = new ServiceCollection();
-                        serviceCollection.AddSignalRCore();
-                        serviceCollection.AddSingleton<IConfigureOptions<HubOptions>, ManagementHubOptionsSetup>();
-
-                        if (loggerFactory != null)
-                        {
-                            serviceCollection.AddSingleton(typeof(ILoggerFactory), loggerFactory);
-                        }
-
-                        serviceCollection
-                            .AddLogging()
-                            .AddSingleton(typeof(IConnectionFactory), sp => connectionFactory)
-                            .AddSingleton(typeof(HubLifetimeManager<>), typeof(WebSocketsHubLifetimeManager<>))
-                            .AddSingleton(typeof(IServiceConnectionManager<>), typeof(ServiceConnectionManager<>))
-                            .AddSingleton(typeof(IServiceConnectionContainer), sp => weakConnectionContainer);
-
```

Comment:
This refactor is necessary to avoid duplication of code. The strongly typed hub shares a lot of functionality.